core/trino-main/src/main/java/io/trino/sql/planner/rowpattern/AggregateArgumentsRewriter.java

```diff
@@ -0,0 +1,150 @@
+/*
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package io.trino.sql.planner.rowpattern;
+
+import com.google.common.collect.ImmutableList;
+import com.google.common.collect.ImmutableSet;
+import io.trino.sql.planner.Symbol;
+import io.trino.sql.planner.SymbolAllocator;
+import io.trino.sql.tree.Expression;
+import io.trino.sql.tree.ExpressionRewriter;
+import io.trino.sql.tree.ExpressionTreeRewriter;
+import io.trino.sql.tree.FunctionCall;
+import io.trino.sql.tree.LabelDereference;
+import io.trino.sql.tree.QualifiedName;
+
+import java.util.List;
+import java.util.Set;
+
+import static io.trino.spi.type.BigintType.BIGINT;
+import static io.trino.spi.type.VarcharType.VARCHAR;
+import static io.trino.sql.analyzer.ExpressionAnalyzer.isPatternRecognitionFunction;
+import static java.util.Locale.ENGLISH;
+import static java.util.Objects.requireNonNull;
+
+public class AggregateArgumentsRewriter
+{
+    private AggregateArgumentsRewriter() {}
+
+    /**
+     * This rewriter is dedicated for aggregation arguments in pattern recognition context.
+     * <p>
+     * Rewrite expressions so that they do not contain any elements specific to row pattern recognition:
+     * - remove labels from `LabelDereference`s and `CLASSIFIER()` calls,
+     * - replace `CLASSIFIER()` and `MATCH_NUMBER()` calls with new symbols,
+     * - the expressions do not contain navigations by analysis.
+     *
+     * @return Rewritten expressions along with sets of allocated symbols which replace `CLASSIFIER()` and `MATCH_NUMBER()` calls.
```

Comment:
Currently the javadoc looks different. I think you might be reviewing some outdated version. Which would be very strange, since I last pushed almost a month ago... Maybe there is some inconsistency in Github?